File: php/testcode/weakcipher_032.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakcipher_blowfish_static_iv
function weakcipher032(BenchmarkRequest $req): BenchmarkResponse {
    $key = random_bytes(16);
    $staticIv = 'staticiv12345678';
    $enc = openssl_encrypt($req->param('data'), 'BF-CBC', $key, 0, $staticIv); // vuln-code-snippet vuln-line php_weakcipher_blowfish_static_iv
    return BenchmarkResponse::ok($enc);
}
// vuln-code-snippet end php_weakcipher_blowfish_static_iv

File: ruby/testcode/weakcipher_032.rb
require_relative 'shared'
require 'openssl'

FIXED_KEY_32 = 'aaaabbbbccccddddeeeeffffgggghhhh'
FIXED_IV_16  = 'aaaabbbbccccdddd'

# vuln-code-snippet start ruby_weakcipher_reused_key_iv
def encrypt_reused_key_iv(req)
  plaintext = req.body_str
  cipher = OpenSSL::Cipher.new('AES-256-CBC')
  cipher.encrypt
  cipher.key = FIXED_KEY_32 # vuln-code-snippet vuln-line ruby_weakcipher_reused_key_iv
  cipher.iv  = FIXED_IV_16
  ciphertext = cipher.update(plaintext) + cipher.final
  BenchmarkResponse.ok(ciphertext.unpack1('H*'))
end
# vuln-code-snippet end ruby_weakcipher_reused_key_iv
